CMPUT301Project/app/src/main/java/com/example/GroupProject/ProfileFragment.java

```diff
@@ -31,24 +60,96 @@ public ProfileFragment() {
      * @return A new instance of fragment ProfileFragment.
      */
     // TODO: Rename and change types and number of parameters
-    public static ProfileFragment newInstance(String param1, String param2) {
+    public static ProfileFragment newInstance(String username) {
         ProfileFragment fragment = new ProfileFragment();
         Bundle args = new Bundle();
+        args.putString("username", username);
         fragment.setArguments(args);
         return fragment;
     }
 
+    @Override
+    public void onAttach(@NonNull Context context) {
+        super.onAttach(context);
+        thisContext = context;
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        FirebaseFirestore db;
+        db = getFirestoreInstance();
+
+        confirmUserbtn.setOnClickListener(view1 -> {
+            Map<String, Object> user = new HashMap<>();
+            if (usernameEdit.getText() != null) {
+                String newUsername = usernameEdit.getText().toString();
+                thisUser.setUsername(newUsername);
+            }
+            if (fnameEdit.getText() != null) {
+                String newFname = fnameEdit.getText().toString();
+                thisUser.setFirstName(newFname);
+            }
+            if (lnameEdit.getText() != null) {
+                String newLname = lnameEdit.getText().toString();
+                thisUser.setLastName(newLname);
+            }
+            user.put("FirstName", thisUser.getFirstName());
+            user.put("LastName", thisUser.getLastName());
+            ((GroupProject) getActivity().getApplication()).setUsername(thisUser.getUsername());
+
+            db.collection("Users").document(thisUser.getUsername())
+                    .set(user, SetOptions.merge());
+        });
+    }
+
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        if (getArguments() != null) {
-        }
+
+        //username = getArguments().getString("username");
+        username = ((GroupProject) getActivity().getApplication()).getUsername();
+        thisUser = new User(username);
+        thisUser.setFirstName("");
```

Comment:
Firstname and lastname should probably be initialized in the constructor.

Reply:
I didn't do that incase someone doesn't want their full name in the app.